internal/controller/dnsrecord_controller.go

```diff
@@ -305,10 +305,22 @@ func (r *DNSRecordReconciler) publishRecord(ctx context.Context, dnsRecord *v1al
 			Namespace: dnsRecord.Namespace,
 		},
 	}
+
 	err := r.Get(ctx, client.ObjectKeyFromObject(managedZone), managedZone, &client.GetOptions{})
 	if err != nil {
 		return false, err
 	}
+
+	patchFrom := client.MergeFrom(dnsRecord.DeepCopy())
+	err = controllerutil.SetOwnerReference(managedZone, dnsRecord, r.Scheme)
+	if err != nil {
+		return false, err
+	}
+	err = r.Client.Patch(ctx, dnsRecord, patchFrom)
+	if err != nil {
+		return false, err
+	}
```

Comment:
Why patch instead of update?

Reply:
I assumed this would help avoid a conflict when the controller tried to update the status later, is that not true?

Reply:
I just realised this is happening way down in `publishRecord`. If we are setting up a relationship between the DNSRecord and the ManagedZone, and this reconciler is the thing ensuring it, i would have expected it be done done at the very start of the Reconcile, with a more explicit step much like when we add the [finalizer](https://github.com/Kuadrant/dns-operator/blob/main/internal/controller/dnsrecord_controller.go#L126).

Issue we have at the moment is that we don't load the MZ until we get down here, but i think we need to change that anyway re. logging [see](https://github.com/Kuadrant/dns-operator/pull/152#discussion_r1626235920)

Generally i think the `Reconcile` function should probably first ensure all bits that require an update to the resource are done and then move on to the actual reconciliation of the spec once that is complete.

1. * Add resource data this reconciler is ensuring (OwnerRefs, finalizers etc..)
2. * Update and return for an immediate re-queue
3. * Process the spec of the resource
4. * Update the resource status and return with validation re-queue

We can leave it like this if we don't want to change that here, but i would expect that to be changed fairly soon after as part of follow on work.